File: server/pictures/PictureAlreadyExistsException.ts
import CatEmployee from "../users/CatEmployee.js";
import PictureFile from "./PictureFile.js";

export class PictureAlreadyExistsException extends Error {
    constructor(pictureFile: PictureFile, catEmployee: CatEmployee) {
        super(`Picture "${pictureFile.fileName}" already exists for user \`${catEmployee.id}\`.`);
    }
}

File: src/main/kotlin/info/davidvedvick/seis739/catpics/pictures/PictureAlreadyExistsException.kt
package info.davidvedvick.seis739.catpics.pictures

import info.davidvedvick.seis739.catpics.users.CatEmployee
import java.io.IOException

class PictureAlreadyExistsException(pictureFile: PictureFile, catEmployee: CatEmployee)
    : IOException("""Picture "${pictureFile.fileName}" already exists for user `${catEmployee.id}`.""")